airflow/serialization/serialized_objects.py

```diff
@@ -256,7 +259,7 @@ def _deserialize(cls, encoded_var: Any) -> Any:  # pylint: disable=too-many-retu
             return SerializedBaseOperator.deserialize_operator(var)
         elif type_ == DAT.DATETIME:
             return pendulum.from_timestamp(var)
-        elif type_ == DAT.POD:
+        elif has_kubernetes and type_ == DAT.POD:
```

Comment:
```suggestion
        elif HAS_KUBERNETES and type_ == DAT.POD:
```

Reply:
Though I also wonder if it should be an error to try and deserialise a pod without the kube modules installed?

So something like:

```suggestion
        elif type_ == DAT.POD:
            if not HAS_KUBERNETES:
                raise RuntimeError("Cannot deserialize POD objects without kubernetes libraries installed!")
```